File: pruebas/SOA/Testeos/Android/Cursor/con formato Prompt/appCursorMQTT/app/src/main/java/com/example/appcursormqtt/ShakeActivity.java
package com.example.appcursormqtt;

import android.content.Context;
import android.hardware.Sensor;
import android.hardware.SensorEvent;
import android.hardware.SensorEventListener;
import android.hardware.SensorManager;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;
import androidx.constraintlayout.widget.ConstraintLayout;

public class ShakeActivity extends AppCompatActivity implements SensorEventListener {

    private ConstraintLayout shakeLayout;
    private Button btnBack;
    private SensorManager sensorManager;
    private Sensor accelerometer;
    
    // Variables para detectar shake
    private long lastUpdate = 0;
    private float last_x, last_y, last_z;
    private static final int SHAKE_THRESHOLD = 600;
    
    // Colores para el fondo
    private int[] colors = {
        R.color.red,
        R.color.black,
        R.color.blue,
        R.color.green,
        R.color.yellow
    };
    private int currentColorIndex = 0;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_shake);
        
        // Inicializar vistas
        shakeLayout = findViewById(R.id.shakeLayout);
        btnBack = findViewById(R.id.btnBack);
        
        // Configurar sensor de acelerómetro
        sensorManager = (SensorManager) getSystemService(Context.SENSOR_SERVICE);
        accelerometer = sensorManager.getDefaultSensor(Sensor.TYPE_ACCELEROMETER);
        
        // Configurar listener del botón
        btnBack.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                finish();
            }
        });
        
        Toast.makeText(this, "Agita el dispositivo para cambiar el color", Toast.LENGTH_LONG).show();
    }
    
    @Override
    protected void onResume() {
        super.onResume();
        // Registrar el listener del sensor
        sensorManager.registerListener(this, accelerometer, SensorManager.SENSOR_DELAY_NORMAL);
    }
    
    @Override
    protected void onPause() {
        super.onPause();
        // Desregistrar el listener del sensor
        sensorManager.unregisterListener(this);
    }
    
    @Override
    public void onSensorChanged(SensorEvent event) {
        if (event.sensor.getType() == Sensor.TYPE_ACCELEROMETER) {
            long curTime = System.currentTimeMillis();
            
            if ((curTime - lastUpdate) > 100) {
                long diffTime = (curTime - lastUpdate);
                lastUpdate = curTime;
                
                float x = event.values[0];
                float y = event.values[1];
                float z = event.values[2];
                
                float speed = Math.abs(x + y + z - last_x - last_y - last_z) / diffTime * 10000;
                
                if (speed > SHAKE_THRESHOLD) {
                    // Shake detectado - cambiar color
                    changeBackgroundColor();
                }
                
                last_x = x;
                last_y = y;
                last_z = z;
            }
        }
    }
    
    @Override
    public void onAccuracyChanged(Sensor sensor, int accuracy) {
        // No necesitamos hacer nada aquí
    }
    
    private void changeBackgroundColor() {
        // Cambiar al siguiente color
        currentColorIndex = (currentColorIndex + 1) % colors.length;
        
        // Aplicar el nuevo color al fondo
        shakeLayout.setBackgroundResource(colors[currentColorIndex]);
        
        // Mostrar toast con el color actual
        String[] colorNames = {"Rojo", "Negro", "Azul", "Verde", "Amarillo"};
        Toast.makeText(this, "Color cambiado a: " + colorNames[currentColorIndex], Toast.LENGTH_SHORT).show();
    }
} 

File: vibecoding/SOA/Testeos/Android/Windsurf/con formato prompt/kotlin/appWindsurfMQTT/app/src/main/java/com/example/appwindsurfmqtt/ShakeActivity.kt
package com.example.appwindsurfmqtt

import android.content.Context
import android.hardware.Sensor
import android.hardware.SensorEvent
import android.hardware.SensorEventListener
import android.hardware.SensorManager
import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import androidx.core.content.ContextCompat

class ShakeActivity : AppCompatActivity(), SensorEventListener {
    private var sensorManager: SensorManager? = null
    private var accelerometer: Sensor? = null
    private var lastUpdate: Long = 0
    private var lastX = 0f
    private var lastY = 0f
    private var lastZ = 0f
    private val shakeThreshold = 800 // Adjust this value to change shake sensitivity
    
    private val colors = listOf(
        android.R.color.holo_red_dark,
        android.R.color.black,
        android.R.color.holo_blue_dark,
        android.R.color.holo_green_dark,
        android.R.color.holo_orange_dark
    )
    private var currentColorIndex = 0

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_shake)
        
        // Initialize the sensor service
        sensorManager = getSystemService(Context.SENSOR_SERVICE) as SensorManager
        accelerometer = sensorManager?.getDefaultSensor(Sensor.TYPE_ACCELEROMETER)
    }

    override fun onResume() {
        super.onResume()
        // Register the sensor listener when the activity is in the foreground
        sensorManager?.registerListener(
            this,
            accelerometer,
            SensorManager.SENSOR_DELAY_NORMAL
        )
    }

    override fun onPause() {
        super.onPause()
        // Unregister the sensor listener when the activity is paused
        sensorManager?.unregisterListener(this)
    }

    override fun onSensorChanged(event: SensorEvent) {
        if (event.sensor.type == Sensor.TYPE_ACCELEROMETER) {
            val currentTime = System.currentTimeMillis()
            
            // Only check for shake every 100ms
            if ((currentTime - lastUpdate) > 100) {
                val timeDiff = currentTime - lastUpdate
                lastUpdate = currentTime
                
                val x = event.values[0]
                val y = event.values[1]
                val z = event.values[2]
                
                val speed = Math.abs(x + y + z - lastX - lastY - lastZ) / timeDiff * 10000
                
                if (speed > shakeThreshold) {
                    // Shake detected, change the background color
                    changeBackgroundColor()
                }
                
                lastX = x
                lastY = y
                lastZ = z
            }
        }
    }
    
    private fun changeBackgroundColor() {
        // Get the next color in the list
        currentColorIndex = (currentColorIndex + 1) % colors.size
        val color = ContextCompat.getColor(this, colors[currentColorIndex])
        
        // Change the background color
        window.decorView.setBackgroundColor(color)
    }

    override fun onAccuracyChanged(sensor: Sensor?, accuracy: Int) {
        // Not needed for this implementation
    }
}
